modules/core/src/main/java/org/apache/fluo/core/oracle/OracleServer.java

```diff
@@ -318,6 +324,10 @@ public synchronized void start() throws Exception {
     curatorFramework.getConnectionStateListenable().addListener(cnxnListener);
     curatorFramework.start();
 
+    while (!curatorFramework.getState().equals(CuratorFrameworkState.STARTED)) {
```

Comment:
I don't think this is needed because curator immediately changes the state in `start()`.  After start returns, the state will be `STARTED`.